Fix issues with rabbitmqctl shell outs; validate fields in resource.

johnbellone opened the pull request:
This patch fixes several issues with the rabbitmqctl shell outs. It also
removes a lot of cruft pushing the validation of parameters down into
the resource instead of in the provider.

Commit:
Fix issues with rabbitmqctl shell outs; validate fields in resource.

This patch fixes several issues with the rabbitmqctl shell outs. It also
removes a lot of cruft pushing the validation of parameters down into
the resource instead of in the provider.

sunggun-yu commented:
@johnbellone 
Thank you for the PR! I'll review and test it.

sunggun-yu commented:
@johnbellone 
It is really great sense to use `rabbitmqctl eval`. I've not tried before. do you have any reference of rabbitmq eval expressions?

johnbellone commented:
@sunggun-yu No reference. I just started digging through the [RabbitMQ code](https://github.com/rabbitmq/rabbitmq-server) and [looked at the Erlang documentation](http://www.erlang.org/doc.html).

sunggun-yu commented:
@johnbellone 
Hi,
Thank you for the information. I have to learn Erlang lol

BTW, I'm still testing PR. I've tested with vagrant box in the project. however, it is not working as expectation.
The `vagrant up` creates 3 vm with clustering. can you test with that? you need merge back from master to yours since I made fix on vagrant file. and it worked partially when I add some log in the `rabbitmqctl_cluster.rb`
Thank you.

johnbellone commented:
Let me give it a whirl.

sunggun-yu commented:
@johnbellone this cookbook has been merged to rabbitmq cookbook. so rabbitmq-cluster cookbook will be deprecated. Thank you.

johnbellone commented:
Instead of deprecating would you mind transferring ownership to me on Supermarket?

sunggun-yu commented:
@johnbellone 
I added you as collaborator and undo deprecation.

Commit:
Fix issue where exitstatus was being squashed.

Commit:
Fix several problems with resource/providers.

There were several problems with my previous attempt at rewriting this
cookbook. After a few tests locally this design seems to be a lot more
straight forward.

Commit:
Add recipes for master and slave instances of RMQ.

Commit:
Update the README a wee bit.

Commit:
Remove the attributes file.

Commit:
Update the license for the cookbook.

Commit:
Update a few configuration files.

Commit:
Add the attributes default file back to project.

Commit:
Fix a problem with tags on the slave instances.

Commit:
Remove requirement as name attribute is always required.

Commit:
Use the plugin management cookbook to install any plugins.

Commit:
Spew some debug information in the provider.

Commit:
Fix the name of the master node to use the fqdn.

Commit:
Adds poise library helper and cleans up HWRP.

Commit:
Modifies attribute validation for node_type to use equal_to.

Commit:
Pins the version of the RabbitMQ cookbook.

Commit:
Adds functionality to manage configure/install of RabbitMQ.

Commit:
Fixes some issues with the convergence and start of service.
